.github/workflows/push_event.yml

```diff
@@ -0,0 +1,34 @@
+name: Quality control
+
+on: [push]
+
+jobs:
+  build:
+    if: github.ref != 'refs/heads/master'
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Cache Composer packages
+        id: composer-cache
+        uses: actions/cache@v2
+        with:
+          path: vendor
+          key: ${{ runner.os }}-node-${{ hashFiles('/composer.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Install dependencies
+        id: dep-install
+        run: composer install --no-progress --no-suggest
+
+      - name: Run check
+        id: qc-check
+        run: |
+          vendor/bin/phpcs
+          vendor/bin/psalm
+          vendor/bin/phpunit
+          vendor/bin/infection --min-msi=99 --min-covered-msi=100
+          vendor/bin/roave-no-leaks
+          mkdir -p /tmp/dependency-check && composer require --dev -d/tmp/dependency-check maglnet/composer-require-checker:^2.1.0 && /tmp/dependency-check/vendor/bin/composer-require-checker check composer.json
```

Comment:
I'd also suggest using a different build dir with a `composer.lock` just for this one

Reply:
Not sure I understood last comment.   
Different build dir for building advisories ? Or smth else instead of `/tmp`   
Also can we "composer install" it ? Why it must be installed and run independently ?

Reply:
We could run it as dev dependency. The reason this wasn't done initially is that it probably held back some dependency upgrades, but I think that is solved now